include/envoy/network/listen_socket.h

```diff
@@ -34,5 +34,65 @@ class ListenSocket {
 typedef std::unique_ptr<ListenSocket> ListenSocketPtr;
 typedef std::shared_ptr<ListenSocket> ListenSocketSharedPtr;
 
+/**
+ * An abstract accept socket.
+ */
+class AcceptSocket {
```

Comment:
I think I would call this `AcceptedSocket`. That's a little more natural for me. Please update related comments also.

Reply:
Done.